htdocs/libraries/icms/messaging/MentionService.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+declare(strict_types=1);
+
+namespace icms\messaging;
+
+use icms_messaging_Handler;
+use PDO;
+use Throwable;
+
+/**
+ * MentionService
+ * - Extract mentioned users from rich text content
+ * - Notify mentioned users via ImpressCMS messaging (PM)
+ *
+ * Conventions
+ * - Parses plain text @username tokens
+ * - Parses legacy HTML links to user profiles containing uid query param
+ * - Resolves only active users (level > 0)
+ */
+final class MentionService
+{
+    /**
+     * Extract unique mentioned user IDs from content.
+     * Supports:
+     * - Plain text: "@username" (alnum, underscore, dash, dot allowed)
+     * - Legacy HTML link: <a href="...uid=123">...</a>
+     */
+    public static function extractMentionedUserIds(string $content): array
+    {
+        $userIds = [];
+
+        // 1) Plain text @username
+        $usernames = self::parseAtUsernames($content);
+        if ($usernames) {
+            $idsFromNames = self::mapUsernamesToIds($usernames);
+            foreach ($idsFromNames as $uid) {
+                $userIds[$uid] = true;
+            }
+        }
+
+        // 2) Legacy HTML links with uid=...
+        foreach (self::parseLegacyUidLinks($content) as $uid) {
+            $userIds[$uid] = true;
+        }
+
+        $ids = array_keys($userIds);
```

Comment:
I thinks everything here could be rewritten as:

```suggestion
        $ids = array_unique([
           ...self::mapUsernamesToIds(self::parseAtUsernames($content)),
           ...self::parseLegacyUidLinks($content),
        ]);
```